Makefile.buildkit

```diff
@@ -18,6 +17,31 @@ veryclean: clean-build
 
 BUILDKIT_DOCKERFILE_FILTER := | sed -e "1s|^\#.*|\# syntax = docker/dockerfile:experimental|" -e "s|^RUN\(.*\)make|RUN --mount=type=cache,target=/root/.cache/go-build\1make|"
 
+# Check that files needed for git-less build are up-to-date
+build-context-update: GIT_VERSION BPF_SRCFILES
+
+# Generic rule for augmented .dockerignore files. For 'Dockerfile' the stem ('%') matches just the 'D'.
+# '.git' can not be ignored in the top level '.dockerignore' as builds directly from the Dockerfiles
+# (e.g., on Docker hub and Quay) depend on '.git' for the git SHA.
+_build/%ockerfile.dockerignore: .dockerignore
+	@-mkdir -p $(dir $@)
+	@cp $< $@
+	@echo ".git" >> $@
+
+# Generic rule for Dockerfiles. For 'Dockerfile' the stem ('%') matches just the 'D'.
+_build/%ockerfile: %ockerfile _build/%ockerfile.dockerignore force
+	@-mkdir -p $(dir $@)
+	@cat $< $(BUILDKIT_DOCKERFILE_FILTER) > $@
+	@-grep "^FROM " $@ | cut -d ' ' -f2 | grep -v "scratch" | xargs -P4 -n1 docker pull
```

Comment:
Not sure the exact intent of this but I found that `quay.io/cilium/cilium-runtime` was being pulled in Step 18/31 of the process, after the `make` invocation. I would have expected it to be pre-pulled earlier.